plugins/polio/js/src/forms/ShipmentForm.tsx

```diff
@@ -1,23 +1,71 @@
 /* eslint-disable camelcase */
-import React, { FunctionComponent } from 'react';
+import React, { FunctionComponent, useEffect, useMemo } from 'react';
 // @ts-ignore
 import { useSafeIntl } from 'bluesquare-components';
 import { Grid } from '@material-ui/core';
-import { Field } from 'formik';
+import { Field, useFormikContext } from 'formik';
 import MESSAGES from '../constants/messages';
 import { useStyles } from '../styles/theme';
-import { DateInput, Select, TextInput } from '../components/Inputs';
+import { DateInput } from '../components/Inputs';
 import { polioVaccines } from '../constants/virus';
 import { MultilineText } from '../components/Inputs/MultilineText';
+import { SingleSelect } from '../components/Inputs/SingleSelect';
+import { DebouncedTextInput } from '../components/Inputs/DebouncedTextInput';
 
 type Props = {
     index: number;
     accessor: string;
+    roundIndex: number;
 };
 
-export const ShipmentForm: FunctionComponent<Props> = ({ index, accessor }) => {
+export const shipmentFieldNames = [
+    'vaccine_name',
+    'po_numbers',
+    'vials_received',
+    'estimated_arrival_date',
+    'date_reception',
+    'reception_pre_alert',
+    'comment',
+];
+
+export const ShipmentForm: FunctionComponent<Props> = ({
+    index,
+    accessor,
+    roundIndex,
+}) => {
     const classes: Record<string, string> = useStyles();
     const { formatMessage } = useSafeIntl();
+    const { values = {} as any, setFieldTouched } = useFormikContext();
+    const fieldValues = useMemo(
+        () => values?.rounds?.[roundIndex].shipments?.[index],
+        [index, roundIndex, values?.rounds],
+    );
+    useEffect(() => {
+        // Using every to be able to break the loop
+        shipmentFieldNames.every(key => {
+            if (fieldValues[key]) {
```

Comment:
same as before, `fieldValues ` can be `undefined`

Reply:
Also you have the same spacing issue here

Reply:
It comes from the default padding from the `spacing` prop. We need to override it, either by dividing both padding top and by removing one entirely.

Reply:
removing the padding doesn't solve the problem. I think the inputs's height might interfere here. Leaving as is for now